src/hssm/register.py

```diff
@@ -0,0 +1,84 @@
+"""Module for registering custom models in HSSM."""
+
+from pprint import pformat
+from typing import Dict
+
+from .defaults import (
+    DefaultConfig,
+    default_model_config,
+)
+
+
+def register_model(name: str, config: DefaultConfig) -> None:
+    """Register a new model in HSSM.
+
+    Parameters
+    ----------
+    name : str
+        Name of the model to register
+    config : DefaultConfig
+        Model configuration dictionary with the following structure:
+        {
+            "response": list[str],  # e.g. ["rt", "response"]
+            "list_params": list[str],  # e.g. ["v", "a", "z", "t"]
+            "choices": list[int],  # e.g. [-1, 1]
+            "description": str,  # Model description
+            "likelihoods": {
+                "kind": {  # One of: analytical, approx_differentiable, blackbox
+                    "loglik": Callable,  # Log-likelihood function
+                    "backend": str | None,  # Optional, e.g. "jax" or "pytensor"
+                    "bounds": dict,  # Parameter bounds
+                    "default_priors": dict,  # Default priors for parameters
+                    "extra_fields": list | None  # Optional extra fields
+                }
+            }
+        }
+    """
+    # Validate required keys
+    required_keys = set(DefaultConfig.__annotations__.keys())
+    missing_keys = required_keys - set(config.keys())
+    if missing_keys:
+        raise ValueError(f"Missing required keys in config: {missing_keys}")
+
+    # TODO: validate provided configs?
+
+    # Register the model
+    default_model_config[name] = config
```

Comment:
We are going to improve on this, but I would add some safeguard so that the original defaults are not accidentally overwritten. We can probably provide a `get_model_defaults` that returns a deepcopy of a `DefaultConfig` object for people to overwrite if they only want to change on certain aspects of an existing model

Reply:
Yep, that's basically it :)